new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAddCluster.java

```diff
@@ -0,0 +1,884 @@
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(configMap)  failed ");
+    assertTrue(cmPatched, "patchDomainCustomResource(configMap) failed");
+    
+    patchStr = new StringBuffer("[{");
+    patchStr.append(" \"op\": \"replace\",")
+        .append(" \"path\": \"/spec/replicas\",")
+        .append(" \"value\": 1")
+        .append(" }]");
+    logger.log(Level.INFO, "Replicas patch string: {0}", patchStr);
+
+    patch = new V1Patch(new String(patchStr));
+    boolean repilcaPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(restartVersion)  failed ");
+    assertTrue(repilcaPatched, "patchDomainCustomResource(repilcas) failed");
+
+    patchStr = new StringBuffer("[{");
+    patchStr.append(" \"op\": \"replace\",")
+        .append(" \"path\": \"/spec/restartVersion\",")
+        .append(" \"value\": \"2\"")
+        .append(" }]");
+    logger.log(Level.INFO, "Restart version patch string: {0}", patchStr);
+
+    patch = new V1Patch(new String(patchStr));
+    boolean rvPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(restartVersion)  failed ");
+    assertTrue(rvPatched, "patchDomainCustomResource(restartVersion) failed");
+
+    // Check if the admin server pod has been restarted 
+    // by comparing the PodCreationTime before and after rolling restart
+    checkPodRestarted(adminServerPodName, domainUid, domainNamespace, adminPodCreationTime);
+
+    // Check if the managed server pods have been restarted 
+    // by comparing the PodCreationTime before and after rolling restart
+    for (int i = 1; i <= replicaCount; i++) {
+      checkPodRestarted(managedServerPrefix + 1, domainUid, 
+           domainNamespace, managedServerPodOriginalTimestampList.get(i - 1));
+    }
+
+    // The ServerNamePrefix for the new dynamic cluster is dynamic-server
+    // Make sure the managed server from the new cluster is running
+    
+    String newServerPodName = domainUid + "-dynamic-server1";
+    checkPodCreated(newServerPodName, domainUid, domainNamespace);
+    checkPodReady(newServerPodName, domainUid, domainNamespace);
+    checkServiceCreated(newServerPodName, domainNamespace);
+
+    boolean isServerConfigured = checkManagedServerConfiguration("dynamic-server1");
+    assertTrue(isServerConfigured, "Could not find new managed server configuration");
+    logger.info("Found new managed server configuration");
+  }
+
+  /**
+   * Create a configmap with a sparse model file to add a configured cluster.
+   * Patch the domain resource with the configmap.
+   * Update the restart version of the domain resource to 3
+   * Patch the domain resource with the spec/replicas set to 1
```

Comment:
Ideally, the operator should bring up all servers in the new cluster concurrently - without 'rolling' them.  I don't know how to check this except perhaps to have the new server be a two server cluster and verify that its second server starts before its first server becomes 'ready'.

Reply:
The usecase makes sure that a new cluster configuration is added to the domain and all the servers are up and running. Since I patch domain resource with the spec/replicas set to 1, only one server from the new cluster comes up.

Reply:
That's my point.  If you bring up two servers, you can test if the operator is bringing them up in parallel - as expected.

Reply:
All the managed servers should be stated in parallel when the managed servers are started in any model.  So this is not an MII specific behavior. We can track the verification of this use case as a generic pod lifecycle test.